### DISCUSSION TITLE
Making an dashboard editable when it is embedded into an application

### DISCUSSION THREAD

**Discussion by kodimalarv** [apache/superset#25208] [Q&A / Help]
As of now we can embed the dashboard into a application, but there is no direct option to edit the dashboard, if there is an option, please give me an example so that it will be easy to use! 

#### How to reproduce the bug
Embedded the dashboard using the Embedded-SDK in a react application, Now its good to view, but If I want to edit the dashboard I have to go to the Superset to make changes, But its easy if the edit dashboard option is available on the embedded app itself. 
![image](https://github.com/apache/superset/assets/134506437/e7a45074-7b05-41f5-82ce-c8ac9868a607)

**Comment by huamichaelchen** [comment]
I suspect if you added the `Public` or `Gamma` role to allow "edit dashboard" could resolve the issue. That's been said, it's probably not a good idea to allow embedded stuff to be edited?

**Comment by sfirke** [comment]
Since this is more of a question or request for help, not a bug report, I'm going to convert it to a discussion.  Hopefully someone will chime in there.  Or, feel free to try asking in the Apache Superset Slack chat - there's a channel there for discussing embedded dashboards.

**Comment by roskee** [comment]
I came across this when I was embedding some dashboards to my frontend. The `edit chart` option is available for each chart on the embedded dashboard but when I click on it it just navigates into a blank white screen. (there are no network requests being made too, when I inspected the dev tools). What am I missing?

**Reply by gauthamhc09** [reply]
Hello roskee, 
Did you find any solution on this ?

**Reply by rohitushur** [reply]
@sfirke @roskee  Is there any solution for this. We are facing this issue and not able to resolve.
@roskee are you still stuck or able to fix this? @gauthamhc09 any suggestions from you to make this work?

**Reply by roskee** [reply]
No. We decided to not use it altogether. I discussed it on their slack channel and if I remember correctly, it is not implemented.

**Reply by ggoouutthhaamm** [reply]
i was able to find the solution, use custom security manager to intercept the login request, then use redirect to redirect to appropriate edit dashboard page.


class CustomAuthDBView(AuthDBView):
    @expose('/login/', methods=['GET', 'POST'])
    def login(self):
        token = request.args.get('token')
        redirect_url = request.args.get('redirect')
        sm = self.appbuilder.sm
        session = sm.get_session
        if not token:
            return super(CustomAuthDBView, self).login()
        try:
            token = decode_token(token)
            id = token["sub"]
            user = session.query(sm.user_model).filter_by(id=id).first()
            login_user(user, remember=False, force=True)
            if not redirect_url:
                return redirect(self.appbuilder.get_url_for_index)
            return redirect(redirect_url)
        except:
            flash('Unable to auto login with access_token', 'warning')
            return super(CustomAuthDBView,self).login()

class CustomSecurityManager(SupersetSecurityManager):
    authdbview = CustomAuthDBView
    #login using access_token or user credentials
    def __init__(self, appbuilder):
        super(CustomSecurityManager, self).__init__(appbuilder) 
        
        
 HTML:
 
 <iframe
          id="ss-iframe"
          sandbox="allow-same-origin allow-scripts allow-forms allow-downloads allow-popups allow-presentation"
          src={url}
          title={title}
        ></iframe>       
        
    <script>    
        const redirectUrl = `${dashboard_url}?standalone=1`;
        url =  `${server_url}/login?token=${loginToken.access_token}&redirect=${redirectUrl}`;
    </script>

**Comment by gauthamhc09** [comment]
No it is not possible. Dashboard's are not editable in the embedded level.

Note* : I found some reply replicating my name, that is not me.

**Reply by manik-ushur** [reply]
@ggoouutthhaamm @gauthamhc09 Do you guys know how to add an api endpoint to superset in superset_config.py? I am adding it using the above code. But getting 404 errors.

**Comment by dax-hurley** [comment]
Hello,

Just wanted to add to this discussion because I've personally been spending a great deal of time trying to find a FOSS BI tool that allows embedding the editor experience and found this discussion while searching. I think this is something that's missing from all FOSS BI and visualization tools currently, even though it is a relatively common feature in commercial apps (which I don't have the $ for since this is for a personal project). @ggoouutthhaamm's solution elsewhere in this thread looks interesting and is something I may explore, but it feels a little "hacky," so I'm not very confident about using it in prod.

I hope this could potentially be considered as a feature request to be officially supported in Superset, as it would be very helpful to myself and others.
